src/main/java/org/jenkinsci/plugins/jiraext/view/FirstWordOfCommitStrategy.java

```diff
@@ -69,13 +69,23 @@ public FirstWordOfCommitStrategy()
         String msg = change.getMsg();
         String firstWordOfTicket;
         firstWordOfTicket = msg.substring(0, (msg.contains(" ") ? StringUtils.indexOf(msg, " ") : msg.length()));
+        firstWordOfTicket = firstWordOfTicket.substring(0, (firstWordOfTicket.contains("]") ? StringUtils.indexOf(firstWordOfTicket, "]") : firstWordOfTicket.length()));
```

Comment:
first word of ticket is meant to hold the first string of characters not containing a space (or the entire commit message, if it contains a space)